Services/Administration/OJS.Services.Administration.Business/Problems/Validators/ProblemAdministrationValidator.cs

```diff
@@ -21,56 +22,65 @@ public class ProblemAdministrationValidator : BaseAdministrationModelValidator<P
 
-            this.RuleFor(model => model.Name)
+        this.RuleFor(model => model.Name)
                 .Length(1, ConstraintConstants.Problem.NameMaxLength)
                 .When(x => x.OperationType is CrudOperationType.Create or CrudOperationType.Update);
 
-            this.RuleFor(model => model.SourceCodeSizeLimit)
+        this.RuleFor(model => model.SourceCodeSizeLimit)
                 .GreaterThanOrEqualTo(0)
                 .WithMessage("Source code size limit cannot be zero or less.")
                 .When(x => x.OperationType is CrudOperationType.Create or CrudOperationType.Update);
 
-            this.RuleFor(model => model.MaximumPoints)
+        this.RuleFor(model => model.MaximumPoints)
                 .GreaterThanOrEqualTo((short)0)
                 .WithMessage("Maximum points cannot be zero or less.")
                 .When(x => x.OperationType is CrudOperationType.Create or CrudOperationType.Update);
 
-            this.RuleFor(model => model.SubmissionTypes.Count)
+        this.RuleFor(model => model.SubmissionTypes.Count)
                 .GreaterThanOrEqualTo(1)
                 .WithMessage("There must be at least one submission type.")
                 .When(x => x.OperationType is CrudOperationType.Create or CrudOperationType.Update);
 
-            this.RuleFor(model => model.ContestId)
+        this.RuleFor(model => model.ContestId)
                 .GreaterThanOrEqualTo(0)
                 .WithMessage("Must select valid contest.")
                 .When(x => x.OperationType is CrudOperationType.Create or CrudOperationType.Update);
 
-            this.RuleFor(model => model.CheckerId)
+        this.RuleFor(model => model.CheckerId)
                 .NotNull()
                 .WithMessage("Checker cannot be null")
                 .GreaterThanOrEqualTo(1)
                 .WithMessage("Checker must be valid type")
                 .When(x => x.OperationType is CrudOperationType.Create or CrudOperationType.Update);
 
-            this.RuleFor(model => model.Id)
+        this.RuleFor(model => model.Id)
                 .MustAsync(async (id, _) => await this.ContestMustNotBeActive(id))
                 .When(x => x.OperationType is CrudOperationType.Delete);
 
-            this.RuleFor(model => model)
+        this.RuleFor(model => model)
                 .CustomAsync(this.MustHaveValidTimeAndMemoryLimits)
                 .When(x => x.OperationType is CrudOperationType.Create or CrudOperationType.Update);
-        }
+
+        this.RuleFor(model => model)
+            .MustAsync(async (model, _) => await this.MustHaveValidProblemGroupId(model))
+            .WithMessage("Invalid value for \"Problem Group Order By\" has been provided.")
+            .WhenAsync(async (x, _) => x.OperationType is CrudOperationType.Create or CrudOperationType.Update &&
+                                    await this.contestsActivityService.IsContestActive(x.ContestId));
```

Comment:
Why this rule is checked only if the contest is Active?

Reply:
Done.